From f4b6251621098a7cb42395ecd848d7c2f3ab0f01 Mon Sep 17 00:00:00 2001
From: lzhong <lzhong@l3-laptop.suse>
Date: Thu, 13 Apr 2017 17:10:59 +0800
Subject: [PATCH] hpsa: limit transfer length to 1MB
Patch-mainline: submitted, waiting for acceptance

The hpsa firmware will bypass the cache for any request larger
than 1MB, so we should cap the request size to avoid any
performance degradation in kernels later than v4.3

This degradation is caused from d2be537c3ba3568acd79cd178327b842e60d035e,
which changed max_sectors_kb to 1280k, but the hardware is able to
work fine with it, so the true fix should be from hpsa driver.

References: bsc#1025461

Signed-off-by: Hannes Reinecke <hare@suse.com>
Acked-by: Yadan Fan <ydfan@suse.com>
---
 drivers/scsi/hpsa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 4c5dd6e..3f362a5 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -943,7 +943,7 @@ static struct scsi_host_template hpsa_driver_template = {
 #endif
 	.sdev_attrs = hpsa_sdev_attrs,
 	.shost_attrs = hpsa_shost_attrs,
-	.max_sectors = 8192,
+	.max_sectors = 1024,
 	.no_write_same = 1,
 };
 
-- 
2.10.2

